py-polars/polars/dataframe/frame.py

```diff
@@ -4773,6 +4780,13 @@ def groupby_dynamic(
             - 'datapoint': Start from the first encountered data point.
             - 'monday': Start the window on the monday before the first data point.
 
+        Returns
+        -------
+        DynamicGroupBy
+            Object you can call ``.agg`` on to aggregate by groups, the result
+            of which will be sorted by `index_column` (as well as by the
```

Comment:
This is not entirely correct. The dataframe will not be sorted by `by`, but within the `by` group, the data will be sorted. A slight nuance.